doc/govuk-components.md

```diff
@@ -9,10 +9,58 @@ The available components and their documentation are exposed by an API at `/temp
 * a [SCSS module](../app/assets/stylesheets/govuk-component) - The styling of the component
-* a Javascript module - no examples yet.
+* a [Javascript module](../app/assets/javascripts/govuk-component) - JS behaviour of the component
 * [Documentation](../app/views/govuk_component/docs) - a `.yml` per component, describing the component and containing fixture data.
 * a [unit test](../test/govuk_component) - testing the component renders correctly based on parameters passed in
 
+## Principles behind components
+
+### Encapsulate common UI
+
+GOV.UK Components aim to hide the complexity of a piece of common UI. Putting the templating logic, HTML, CSS and JS behind an abstraction, so that an app rendering a component only needs to know about the parameters to render it with. This is similar to how [React components](https://facebook.github.io/react/index.html) work.
+
+Think of it as a function call in a programming language, as long as the [interface](https://en.wikipedia.org/wiki/Application_programming_interface) doesn't change, then code rendering a component doesn't need to care about the changes.
+
+Here's an example of calling a component. The parameter is `title` with a value of `"My page title"`.
+
+```ruby
+<%= render partial: 'govuk_component/title', locals: { title: "My page title" } %>
+```
+
+This will output something like:
+
+```html
+<div class="govuk-title length-average">
+  <h1>My page title</h1>
+</div>
+```
+
+Apps should treat the output of a component like a [black box](https://en.wikipedia.org/wiki/Black_box), not relying on the internal implementation as it may change (eg, additional CSS based on the `govuk-title` or `h1` selector)
+
+If additional behavior in the component is needed then it should be made in the component itself, then used in an application.
+
+### Distribute via the network
+
+One of the goals of GOV.UK Components is to have consistency between the many different frontend applications. Because of this components live in a central place (the `static` application) and are distributed to the different frontends over the network. This is done via Slimmer, in a similar way to [layout templates](slimmer_templates.md).
+
+This allows a component to be changed and deployed within `static` and each frontend application will automatically fetch a new version of the [component template code]((../app/views/govuk_component).
+
+The automatic updating allows changes to be made to every GOV.UK frontend app, without having to manually update and deploy them. With 20+ frontend apps, this makes updating components much easier.
+
+However, this approach does have some costs. It's less explicit, and not as clear how the common component code is included or updated. Changes to the component interface/parameters must be backwards compatible, which can require multi-step changes, but with established UI patterns this should be less of an issue.
```

Comment:
Yep, just like that - incidentally, thats how most GOV.UK do DB migrations, so it's a familiar pattern.

Your bullet list makes it much clearer, so i've incorporated that into the doc :)